### ISSUE TITLE
Application troubleshooting

### ISSUE THREAD

**Issue by btat** [SUSE/doc-cap#228]
- Get a shell with the app/buildpack env (for any buildpack). Run for each ssh session.
```
cf ssh APP_NAME
exec /tmp/lifecycle/launcher /home/vcap/app "bash -l" ""
```

**Comment by btat** [comment]
Symptoms:
   - cf app using mongodb service deployed by minibroker are failing
     to reach/use the service

# Session

   1. Got access details for AWS jump box which is set up to talk to
      the EKS cluster, at both kube and CF levels (helm, kubectl, cf).

   1. Example application `nodecellar` (org SUSE, space DEV).

   1. `cf ssh nodecellar` (app container), and looked at environment
      variable `VCAP_SERVICES`.

   1. Service host is `wrinkled-marsupial-mongodb.minibroker.svc.cluster.local`
   1. Port is 27017.

   1. `dig wrinkled-marsupial-mongodb.minibroker.svc.cluster.local`
      (still in the app cointainer) resolves to 10.100.49.73

   1. At the kube level the pod `wrinkled-...` is found in namespace `minibroker`.

   1. `kubectl get services --namespace minibroker |grep wrinkled`
      reports matching port, and cluster ip.

   1. Entering the pod ( `kubectl exec -it wrinkled-marsupial-mongodb-5ddfbd594d-q45bc --namespace minibroker -- env TERM=xterm COLUMNS=185 LINES=47 /bin/bash` ) and inspecting

   1. `ps auxww` shows mongodb running.

```
       1001  [...] /opt/bitnami/mongodb/bin/mongod --config /opt/bitnami/mongodb/conf/mongodb.conf
```

   1. No regular tools for network inspection (i.e. no `netstat`, `nc`, `ss`).

   1. `cat /proc/self/net/tcp` yields data. First line shows that
      something is listening on port 6989 ... That is hex !! In
      decimal this is 27017. Our port.

      Conclusion: Mongodb is running and properly listening.

   1. `curl` is available (surprise).

   1. `curl -v http://localhost:27017` ... Works. Mongodb responds,
      complains that we are trying HTTP on the native driver port.

```
       [...]
       It looks like you are trying to access MongoDB over HTTP on the native driver port.
```

   1. `curl -v http://10.100.49.73:27017` ... Also works. Note, the
      difference is that we are using the cluster ip, so the
      connection goes a bit out of the container and then back to
      it. The `localhost` curl was purely in the container.

   1. Leaving the pod/container and trying curl again, i.e.
      `curl -v http://10.100.49.73:27017` ... This hangs.

```
       * Rebuilt URL to: http://10.100.49.73:27017/
       * Hostname was NOT found in DNS cache
       *   Trying 10.100.49.73...
       ^C
```

      It seems that the port (or ip/port) is not reachable from fully
      outside the container itself.

      Theory: Some associated AWS/EKS level security group does not
      allow connections to port 27017.

      Note, we are not talking about the CF level security group
      here. The AWS/EKS SG is lower than that.

**Comment by btat** [comment]
`kubectl edit --namespace uaa statefulsets uaa`
setting `imagePullPolicy: Always` (temporarily) for a statefulset as above seems to fix the `Error: Manifest does not match provided manifest digest`
